partner_contact_age_range/models/res_partner.py

```diff
@@ -0,0 +1,30 @@
+# Copyright (C) 2019-Today: Druidoo (<https://www.druidoo.io>)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models
+
+
+class ResPartner(models.Model):
+    _inherit = "res.partner"
+
+    age_range_id = fields.Many2one(
+        "res.partner.age.range", "Age Range", compute="_compute_age_range", store=True
+    )
+
+    @api.depends("age")
+    def _compute_age_range(self):
+        for record in self:
+            age_range = self.env["res.partner.age.range"].search(
+                [("age_from", "<=", record.age), ("age_to", ">=", record.age)]
+            )
```

Comment:
I'm concerned about usability here.
This will end up in lots of queries (one per partner).

Maybe you can fetch all date ranges before looping through the partners, and then filter on them in python to be more efficient?

Date ranges are probably just a few in any scenario, while on the other hand there could be thousands of partners in a database.